pkg/cmd/server/start.go

```diff
@@ -770,33 +195,40 @@ func env(key string, defaultValue string) string {
 	return val
 }
 
-func expandImage(component, defaultValue string, latest bool) string {
-	template := defaultValue
-	if s, ok := imageComponentEnvExpander(component); ok {
-		template = s
-	}
-	value, err := variable.ExpandStrict(template, func(key string) (string, bool) {
-		switch key {
-		case "component":
-			return component, true
-		case "version":
-			if latest {
-				return "latest", true
-			}
-		}
-		return "", false
-	}, variable.Versions)
+func (cfg Config) RunEtcd() error {
+	systemAddresses, err := cfg.BuildSystemAddresses()
 	if err != nil {
-		glog.Fatalf("Unable to find an image for %q due to an error processing the format: %v", err)
+		return err
+	}
+
+	etcdConfig := &etcd.Config{
+		BindAddr:     getHostOfHost(systemAddresses.MasterBindHost),
+		PeerBindAddr: getHostOfHost(systemAddresses.MasterBindHost),
+		MasterAddr:   systemAddresses.EtcdURL.Host,
+		EtcdDir:      cfg.EtcdDir,
 	}
-	return value
+
+	etcdConfig.Run()
+
+	return nil
 }
 
-func imageComponentEnvExpander(key string) (string, bool) {
-	s := strings.Replace(strings.ToUpper(key), "-", "_", -1)
-	val := os.Getenv(fmt.Sprintf("OPENSHIFT_%s_IMAGE", s))
-	if len(val) == 0 {
-		return "", false
+func getHost(theURL url.URL) string {
+	parts := strings.Split(theURL.Host, ":")
```

Comment:
use `net.SplitHostPort`, and handle an error condition by just returning `theUrl.Host`. same below for hostofhost and getport